pddlgym/core.py

```diff
@@ -155,7 +186,8 @@ def _check_struct_for_strips(struct):
         return all(_check_struct_for_strips(l) for l in struct.literals)
     return False
 
-def _apply_effects(state, lifted_effects, assignments):
+
+def _apply_effects(state, lifted_effects, assignments, get_all_transitions=False):
```

Comment:
This function is messed up, it still needs some refactoring after these new changes.

Reply:
Cool, I'll take a look after you've finished the refactoring.

Reply:
Finished refactoring. Let me know if you have any concerns :)

Reply:
Looks good to me! Please make sure tests pass, both the current ones and the one(s) that you add. After you fix the get_successor_states() issue above, we should be ready to merge.

Reply:
Fine! I've already added one test for this new functionality, did you see it?

Reply:
That test looks good, added a minor comment for it.